### PR TITLE
[Hotfix][ENG-3707] PLOS Metrics

### PR DESCRIPTION
**Pull request by mfraezz**
## Purpose
Add cron command for gathering requested metric data

## Changes
- Add `osf.management.commands.cumulative_plos_metrics`
- Update CeleryConfig

## QA Notes
N/A

## Deployment Notes
- Add `cumulative_plos_metrics` to `beat_schedule` dict

## Side Effects
None Expected

## Ticket

[ENG-3707](https://openscience.atlassian.net/browse/ENG-3707)

### CODE REVIEW

File: osf/management/commands/cumulative_plos_metrics.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+from addons.osfstorage.models import OsfStorageFile
+from framework.celery_tasks import app as celery_app
+from osf.metrics import PreprintView, PreprintDownload
+from osf.models import Guid, Node, OSFUser, Preprint, \
+    Registration, TrashedFile
+from osf.utils.fields import ensure_str
+from website.settings import KEEN, \
+    PLOS_METRICS_BASE_FOLDER, \
+    PLOS_METRICS_OSF_TOKEN, \
+    PLOS_METRICS_INITIAL_FILE_DOWNLOAD_URL
```
Comment (jwalz) [inline osf/management/commands/cumulative_plos_metrics.py:21]:
```suggestion
from website.settings import (
    KEEN,
    PLOS_METRICS_BASE_FOLDER,
    PLOS_METRICS_INITIAL_FILE_DOWNLOAD_URL,
    PLOS_METRICS_OSF_TOKEN,
)
```

Comment (jwalz) [reply osf/management/commands/cumulative_plos_metrics.py:21]: (the nittiest of nits)

File: osf/management/commands/cumulative_plos_metrics.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+
+DEFAULT_API_VERSION = '2.14'
```
Comment (jwalz) [inline osf/management/commands/cumulative_plos_metrics.py:24]: why this version? isn't 2.20 the latest?

File: osf/management/commands/cumulative_plos_metrics.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+def parse_base_file():
+    r = requests.get(
+        url=PLOS_METRICS_INITIAL_FILE_DOWNLOAD_URL,
+        headers={'Content-type': 'application/CSV'},
+        auth=bearer_token_auth(PLOS_METRICS_OSF_TOKEN)
+    )
+    csvr = csv.DictReader(ensure_str(r.content).splitlines(), fieldnames=COL_HEADERS)
+    headers = next(csvr)
+    for k, v in headers.items():
+        assert k == v, f'Unexpected header row: {k} and {v} do not match'
```
Comment (jwalz) [inline osf/management/commands/cumulative_plos_metrics.py:68]:
```suggestion
    csvr = csv.DictReader(ensure_str(r.content).splitlines())
    assert csvr.fieldnames == COL_HEADERS, f'Unexpected headers: expected {COL_HEADERS}, got {csvr.fieldnames}'
```

**Review by jwalz — APPROVED** [review]
Couple nits, but lgtm

File: osf/management/commands/cumulative_plos_metrics.py (hunk 3 above)
Comment (mfraezz) [reply osf/management/commands/cumulative_plos_metrics.py:68]: I considered this, but order matters, and I wasn't certain this comparison guarantees that.

Comment (jwalz) [reply osf/management/commands/cumulative_plos_metrics.py:68]: Raw list and/or tuple comparison requires shared ordering. Also confirmed that `csvr.fieldnames` returns a list.
```
>>> [1, 2, 3, 4, 5] == [1, 2, 3, 5, 4]
False
```

File: osf/management/commands/cumulative_plos_metrics.py (hunk 2 above)
Comment (mfraezz) [reply osf/management/commands/cumulative_plos_metrics.py:24]: Our other scripts that do remote OSF file ops also specify `2.14`. I'm not certain it matters, but there may have been breaking changes between `2.14` and `2.20`

Comment (mfraezz) [reply osf/management/commands/cumulative_plos_metrics.py]: Confirmed `2.20` works as expected

**Review by jwalz — APPROVED** [review]

**Pull request merged**